[nativert] move executor config to torch

dolpm opened the pull request:
Summary:
nativert RFC: https://github.com/zhxchen17/rfcs/blob/master/RFC-0043-torch-native-runtime.md

To land the runtime into PyTorch core, we will gradually land logical parts of the code into the Github issue and get each piece properly reviewed.

This diff moves the executor config to torch. since it's header-only this requires some changes to the libtorch build configs

Test Plan: CI

Differential Revision: D74278789

zhxchen17 commented on build_variables.bzl:
```diff
@@ -592,6 +592,10 @@ libtorch_nativert_sources = [
     "torch/nativert/graph/TensorMeta.cpp",
 ]
 
+libtorch_nativert_standalone_headers = [
+    "torch/nativert/executor/ExecutorConfig.h",
+]
+
```
Not an expert on our build infra but I feel we don't need to special handle headers in the cmake?

Usually we have a catch-all rule to automatically the headers.

zhxchen17 replied:
Oh probably for buck I ran into a similar issue and I fixed in with `//caffe2:libtorch` which should include all the headers now, see [D74184245](https://www.internalfb.com/diff/D74184245)

Skylion007 commented:
@dolpm Please make sure this new nativert folder is covered by clang-tidy

cyyever commented:
> @dolpm Please make sure this new nativert folder is covered by clang-tidy (including the header include regex) in both .lintrunner and .clang-tidy

And of course covered by clang-format.

albanD commented on build_variables.bzl:
```diff
@@ -624,7 +628,7 @@ libtorch_lite_cmake_sources = sorted(
 )
 
-libtorch_cmake_sources = libtorch_core_sources + libtorch_core_jit_sources + libtorch_nativert_sources
+libtorch_cmake_sources = libtorch_core_sources + libtorch_core_jit_sources + libtorch_nativert_sources + libtorch_nativert_standalone_headers
```
What does adding headers to this do? These are sources to be compiled?

zhxchen17 replied:
@albanD I don't think these are source files.

albanD replied:
These are all `.cpp` files is what I meant.

albanD commented on torch/nativert/executor/ExecutorConfig.h:
```diff
@@ -0,0 +1,21 @@
+  bool enableStaticMemoryPlanning = false;
+  bool runConstFolding = false;
+  // allows up to max number of concurrent threads.
+  int64_t maxNumConcurrentThreads = 8;
```
Why not use the at flag we already have for this?

albanD replied:
https://github.com/pytorch/pytorch/blob/fb85ebd71028770df2be314ef8f95acc84306cd3/aten/src/ATen/Parallel.h#L16-L20

albanD commented on torch/nativert/executor/ExecutorConfig.h:
```diff
@@ -0,0 +1,21 @@
+  std::string modelName = "unknown";
+};
+
+} // namespace facebook::sigmoid
```
needs updating

albanD reviewed (commented):
Change of build config looks suspicious.

zhxchen17 commented:
I guess I will stamp here because this is just a config file. Feel free to request changes if there's more things to be addressed.

albanD reviewed (commented):
Setting 8 as default concurrency sounds not ok for sure.

dolpm committed:
[nativert] move executor config to torch (#153087)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/153087

nativert RFC: https://github.com/zhxchen17/rfcs/blob/master/RFC-0043-torch-native-runtime.md

To land the runtime into PyTorch core, we will gradually land logical parts of the code into the Github issue and get each piece properly reviewed.

This diff moves the executor config to torch. since it's header-only this requires some changes to the libtorch build configs

Test Plan: CI

Reviewed By: zhxchen17

Differential Revision: D74278789
